### PR TITLE
feat: add new spinner singleton instance and service for cli commands

### PR DESCRIPTION
**Pull request by larcin**

### CODE REVIEW

**Comment by adriguy** [comment]
@larcin you need to reword your commit: "*feat*" instead of "feature" to make the tests pass 🙂

File: src/services/spinner.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    this.spinnies.fail(this.currentSpinnerUniqueKey, options);
+    this.stop();
+  }
+
+  pause() {
+    if (!this.isRunning()) {
+      throw Error('No spinner is running.');
+    }
+
+    this.spinnies.stopAll();
```
Comment (adriguy) [inline src/services/spinner.js:72]:
```suggestion
    this.spinnies.remove(this.currentSpinnerUniqueKey);
```

File: src/services/spinner.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    if (!this.pausedSpinnerOptions) {
+      throw Error('No spinner has been paused.');
+    }
+
+    this.spinnies.add(this.currentSpinnerUniqueKey, this.pausedSpinnerOptions);
+    this.pausedSpinnerOptions = null;
+  }
+
+  isRunning() {
+    return !!this.currentSpinnerUniqueKey;
```
Comment (adriguy) [inline src/services/spinner.js:90]:
```suggestion
    return !!this.spinnies.pick(this.currentSpinnerUniqueKey);
```

File: src/services/spinner.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+};
+// NOTICE: Singleton used here to attach all generated spinner to the same spinnies instance.
+const spinniesInstance = singletonGetter(Spinnies, spinniesConstructorParameters);
+
+// NOTICE: Except if you need several spinner runnign at the same time,
+//         a `singleton-getter` usage should be prefered.
+/**
+ * A single instance of spinner is intended to be used as follow:
+ * @example
+ * const singletonGetter = require('path/to/singleton-getter')
```
[the remark this answers is not in the document]
Comment (larcin) [reply src/services/spinner.js:20]: As we're already inside the `spinner` file, I thought it was unnecessary

[the remark this answers is not in the document]
Comment (adriguy) [reply src/services/spinner.js:20]: Right!

**Pull request merged**